### DISCUSSION TITLE
Unexpected behaviour while using Query with Depends

### DISCUSSION THREAD

**Discussion by matmalinowski** [fastapi/fastapi#6275] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Depends, Query
from pydantic import BaseModel
from typing import Union

ALIAS_NAME = "param-with-alias"

class ParamsWithoutAlias(BaseModel):

	param_without_alias: Union[str, None] = Query(None)

class ParamsWithAlias(BaseModel):

	param_with_alias: Union[str, None] = Query(None, alias=ALIAS_NAME)

app = FastAPI()

@app.get("/api/get")
async def get_route(
        params: ParamsWithoutAlias = Depends(),
	alias_params: ParamsWithAlias = Depends(),
):

	return {}
```

### Description

While using query params as dependencies, I came across a very interesting problem. The example code above should generate the following OpenApi schema:

![expected_behaviour](https://user-images.githubusercontent.com/36893803/183673315-2431434e-ceb3-456f-9c57-fef70ce65eee.png)

Unfortunately, the ParamsWithAlias class generates the **extra_data** parameter:

![real_behaviour](https://user-images.githubusercontent.com/36893803/183674159-ca2d5cf7-5ed4-464a-b21c-ad4c20301720.png)

Strangely enough, if I change the **ALIAS_NAME** variable to let's say `ALIAS_NAME="paramwithalias"` or `ALIAS_NAME="param_with_alias"` the error doesn't occur:

![paramwithalias](https://user-images.githubusercontent.com/36893803/183674960-aff20295-bc54-41b4-b36f-061b8c9586f2.png)

I also tried using the pydantic **alias_generator** Config method or `pydantic.Field` objects insted of `fastapi.Query`, but it didn't solve the issue either.

### Operating System

Linux

### Operating System Details

Fastapi is being run with uvicorn in docker container with base image **python:3.10-slim-buster**.

### FastAPI Version

0.79.0

### Python Version

Python 3.10.6

**Comment by JarroVGIT** [comment]
This is behaviour that comes from Pydantic, not FastAPI. Whenever you give a 'non-pythonic' alias (like with dashes), the signature that `inspect.signature()` will get from that `BaseModel` subclass will be grouped in the kwargs `extra_data`. A reproducible bit if you want to raise this in the pydantic repo is this:

```python
import inspect
from typing import Union

from fastapi import Query
from pydantic import BaseModel, Field


class A(BaseModel):
    param_with_alias: Union[str, None] = Field(None, alias="hello-you")
    some_other_param_wihouth_alias: str = Field(None)


signature = inspect.signature(A)

print(signature)
```

There is nothing FastAPI can do about this.

**Comment by matmalinowski** [comment]
Thanks a lot for your comment. It is indeed a pydantic issue. I solved it by using dataclasses instead.

**Comment by JarroVGIT** [comment]
Could you post your solution? Might be helpful for future readers that run into the same problem

**Comment by matmalinowski** [comment, marked as answer]
As stated before, if we want to use `fastapi.Query` with alias as a dependency, we should use dataclasses insted of pydantic (due to pydantic issue). So we transform the above code into:

```py
from fastapi import FastAPI, Depends, Query
from dataclasses import dataclass
from typing import Union

ALIAS_NAME = "param-with-alias"

@dataclass
class ParamsWithoutAlias:

	param_without_alias: Union[str, None] = Query(None)

@dataclass
class ParamsWithAlias:

	param_with_alias: Union[str, None] = Query(None, alias=ALIAS_NAME)

app = FastAPI()

@app.get("/api/get")
async def get_route(
        params: ParamsWithoutAlias = Depends(),
	alias_params: ParamsWithAlias = Depends(),
):

	return {}
```

This approach might be useful if we want to create dynamic query params. Example below:

```py
from fastapi import FastAPI, Depends, Query
from dataclasses import dataclass, make_dataclass
from typing import Union

@dataclass
class ExampleQueryParam:

	pass

def create_dynamic_query_params(*args) -> ExampleQueryParam:
	
	return make_dataclass(
		'C',
		fields=list(
			map(
				lambda arg: (arg, Union[str, None], Query(None, alias=f"random-prefix-{arg}")),
				args
			)
		),
		bases=(ExampleQueryParam,)
	)

app = FastAPI()

# The route below will contain param1 and param2 Query params,
# aliased respectively as "random-prefix-param1" and "random-prefix-param2". 
# Both params will be automatically documented in the OpenApi schema.
@app.get("/api/get")
async def route_with_dynamic_params(
        dynamic_params: create_dynamic_query_params("param1", "param2") = Depends()
):
	
	# Accessing params
	param1 = dynamic_params.param1

	return {}
```
